internal/pkg/describe/service.go

```diff
@@ -108,24 +113,35 @@ func newServiceStackDescriber(opt NewServiceConfig, env string) (*serviceStackDe
 
 // Params returns the parameters of the service stack.
 func (d *serviceStackDescriber) Params() (map[string]string, error) {
+	if d.params != nil {
+		return d.params, nil
+	}
 	descr, err := d.cfn.Describe()
 	if err != nil {
 		return nil, err
 	}
+	d.params = descr.Parameters
 	return descr.Parameters, nil
 }
 
 // Params returns the outputs of the service stack.
 func (d *serviceStackDescriber) Outputs() (map[string]string, error) {
+	if d.outputs != nil {
```

Comment:
just to check, these functions with a cache are only called from a single goroutine, right?